o2g/osm/builders/stop_builder.py

```diff
@@ -6,19 +6,77 @@
 
 def build_stops(relations, nodes):
     visited_stops_ids = set()
+    stop_to_station_map = {}
+
+    # First process all stop_areas
     for rel in relations.values():
-        for stop in extract_stops(rel, nodes, visited_stops_ids):
+        if rel.tags.get('public_transport') != 'stop_area':
+            continue
+
+        station = build_parent_stop(rel, nodes)
+        if not station:
+            continue
+
+        for _, member_id, _ in rel.member_info:
+            stop_to_station_map[member_id] = station.stop_id
+
+        visited_stops_ids.add(station.stop_id)
+        yield station
+
+
+    for rel in relations.values():
+        if rel.tags.get('public_transport') == 'stop_area':
+            continue
+
+        for stop in extract_stops(rel, nodes, visited_stops_ids, stop_to_station_map):
             if stop:
                 yield stop
 
 
-def extract_stops(relation, nodes, visited_stop_ids):
+def build_parent_stop(relation, nodes):
+    # One stop per stop_area is necessary.
+    station_node = None
+    some_node = None
+    for member_type, member_id, member_role in relation.member_info:
+        if member_id not in nodes:
+            continue
+        some_node = nodes[member_id]
+        if not some_node:
+            raise Exception()
+        if some_node.tags.get('public_transport') == 'station' or \
+            some_node.tags.get('amenity') == 'bus_station':
+            station_node = some_node
+
+    if not station_node and not some_node:
+        print('No parent node found: https://www.openstreetmap.org/relation/%s' % relation.id)
```

Comment:
is this really necessary? travis ci was complaining about our logs being too much to show :(
I think the case of a stop not having a parent station should be pretty standard

Reply:
oh, I'll clean them up!